### PR TITLE
Add 'No data' state for historical detectors

### PR DESCRIPTION
**Pull request by ohltyler**
*Issue #, if available:* #361

*Description of changes:*

This PR adds a 'No data' state to the historical detectors, to make it more clear to the user when a historical detector's detection date range doesn't return any results / no data to aggregate over. Previously, this error was captured under the vague 'Unexpected failure' detector state, and provided no actionable steps for the user to try to fix it.

Specific changes:
1. Adds the 'No data' state to the list of detector states, only displays specifically for historical detectors
2. Adds a callout with a button to edit the historical detector when in the 'No data' state
3. Adds corresponding UT

Screenshots:
image

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ohltyler** [comment]
Will confirm with UX on the changes before merging.

File: server/routes/utils/adHelpers.ts [hunk 1]
```diff
@@ -325,15 +325,18 @@ export const appendTaskInfo = (
   return detectorMapWithTaskInfo;
 };
 
-// Four checks/transformations need to be made here:
+// Five checks/transformations need to be made here:
```
Comment (yizheliu-amazon) [inline server/routes/utils/adHelpers.ts:328]: minor: change `Five` to `Following`, so that we don't need to change it every time we have new checks.

**Review by yizheliu-amazon — APPROVED** [review]

File: server/routes/utils/adHelpers.ts (hunk 1 above)
Comment (ohltyler) [reply server/routes/utils/adHelpers.ts]: Agreed, changed.

File: server/utils/constants.ts [hunk 2]
```diff
@@ -69,6 +69,7 @@ export enum DETECTOR_STATE {
   FEATURE_REQUIRED = 'Feature required',
   INIT_FAILURE = 'Initialization failure',
   UNEXPECTED_FAILURE = 'Unexpected failure',
+  NO_DATA = 'No data',
```
Comment (ylwu-amzn) [inline server/utils/constants.ts:72]: Not related with this PR. But we may need to test more edge cases like not enough data, exceed our max running AD task limit etc.

**Review by ylwu-amzn — APPROVED** [review]

File: server/utils/constants.ts (hunk 2 above)
Comment (ohltyler) [reply server/utils/constants.ts:72]: Yeah, agreed. I think this "no data" case can be pretty common, so definitely good to have it separated here, but definitely doesn't cover all cases that would currently fall under the 'Unexpected failure' state. I'm assuming those other cases would fall under the `FAILED` state from the backend, and it's probably a matter of parsing out the error message to categorize into other states similar to what's done here.

Comment (ylwu-amzn) [reply server/utils/constants.ts:72]: Yes, we can also parse the error message for "FAILED" state. But not so easy for exceptions which has no readable error message.
